shell/platform/darwin/common/framework/Source/FlutterNSBundleUtils.mm

```diff
@@ -54,7 +54,7 @@
 }
 
 NSString* FLTAssetPath(NSBundle* bundle) {
-  return [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"] ?: @"flutter_assets";
+  return [bundle objectForInfoDictionaryKey:@"FLTAssetsPath"] ?: kDefaultAssetPath;
```

Comment:
This is actually going to fail the g3 roll again. It turned out I was wrong about changing the API to using NSString fixing the issue. The root cause is when the full relative path is used, unloaded-bundle returns a nil path for assets. we have to use "flutter_assets" directly for unloaded-bundles. 

Because I reverted to @"flutter_assets" in https://github.com/flutter/engine/pull/46073, the roll was fixed.

The fix should be trying to load from kDefaultAssetPath first, if not successful, then try @"flutter_assets" for unloaded bundles. 

I will create a PR to fix this with an integration test. 

cc @stuartmorgan @XilaiZhang

Reply:
looks like this was already rolled in by @CaseyHillers

Reply:
Interesting. Maybe it works for XCUITests but not for app extension? That's good to know, we need to fix it for app extensions anyway